src/details/ArborX_Box.hpp

```diff
@@ -25,79 +25,85 @@ namespace ArborX
  * size 2x spatial dimension with a default constructor to initialize
  * properly an "empty" box.
  */
-struct Box
+template <int DIM = 3>
+struct BoxD
 {
-  KOKKOS_DEFAULTED_FUNCTION
-  constexpr Box() = default;
+  static int const dim = DIM;
```

Comment:
Not used.  We'll add it later if we actually need it.

Reply:
I was thinking about deducing the dimension of the data from the return type of the `AccessTraits`. But sure, can be added later.